service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalog.java

```diff
@@ -232,7 +222,16 @@ public String name() {
   }
 
   @VisibleForTesting
-  public FileIO getIo() {
+  public synchronized FileIO getIo() {
+    if (catalogFileIO == null) {
+      LOGGER.debug(
+          "Initializing a default catalogFileIO with properties {}", tableDefaultProperties);
```

Comment:
This is a carry-on from the deleted code. We can remove it if people think it's not useful.